sycl/test-e2e/Basic/parallel_for_range_roundup.cpp

```diff
@@ -1,152 +1,163 @@
 // REQUIRES: gpu
 // RUN: %{build} -o %t.out
-// RUN: env SYCL_PARALLEL_FOR_RANGE_ROUNDING_TRACE=1 %{run} %t.out | FileCheck %s
-
+// RUN: env SYCL_PARALLEL_FOR_RANGE_ROUNDING_TRACE=1 %{run} %t.out | FileCheck %s --check-prefix=CHECK-DEFAULT
+
+// RUN: %{build} -fsycl-range-rounding=force -o %t.out
+// RUN: env SYCL_PARALLEL_FOR_RANGE_ROUNDING_TRACE=1 %{run} %t.out | FileCheck %s --check-prefix=CHECK-DEFAULT
+
+// These tests test 3 things:
+//
+// 1. The user range is the same as the in kernel range (using BufRange) as
+//    reported by get_range().
+// 2. That the effective range is the same as the reported range (using
+//    BufCounter). ie check that the mapping of effective range to user range is
+//    "onto"?
+// 3. That every index in a 1, 2, or 3 dim range is active the execution (using
+//    BufIndexes). ie check that the mapping of effective range to user range is
+//    "1-to-1"?
+//
 #include <iostream>
 #include <sycl/sycl.hpp>
+
 using namespace sycl;
 
+constexpr size_t MagicY = 33, MagicZ = 64;
+
 range<1> Range1 = {0};
 range<2> Range2 = {0, 0};
 range<3> Range3 = {0, 0, 0};
 
+template <typename T> class Kernel1;
+template <typename T> class Kernel2;
+template <typename T> class Kernel3;
+
 void check(const char *msg, size_t v, size_t ref) {
   std::cout << msg << v << std::endl;
   assert(v == ref);
 }
 
-int try_item1(size_t size) {
-  range<1> Size{size};
-  int Counter = 0;
-  {
-    buffer<range<1>, 1> BufRange(&Range1, 1);
-    buffer<int, 1> BufCounter(&Counter, 1);
-    queue myQueue;
-
-    myQueue.submit([&](handler &cgh) {
-      auto AccRange = BufRange.get_access<access::mode::read_write>(cgh);
-      auto AccCounter = BufCounter.get_access<access::mode::atomic>(cgh);
-      cgh.parallel_for<class PF_init_item1>(Size, [=](item<1> ITEM) {
-        AccCounter[0].fetch_add(1);
-        AccRange[0] = ITEM.get_range(0);
-      });
-    });
-    myQueue.wait();
-  }
-  check("Size seen by user = ", Range1.get(0), size);
-  check("Counter = ", Counter, size);
-  return 0;
+template <unsigned Dims> void checkVec(vec<int, Dims> a, vec<int, Dims> b) {
+  assert(Dims == 1 || Dims == 2 || Dims == 3);
```

Comment:
Would use of `check` be better here? If not, at least add an `&& "message" clause to the assertion predicate expression

Reply:
Thanks change made. Have used static assert with message